webui/src/pages/repositories/repository/pulls/pullDetails.jsx

```diff
@@ -101,36 +111,56 @@ const PullDetailsContent = ({repo, pull}) => {
+                {formattedDiffError && <AlertError error={formattedDiffError}/>}
+                {isPullOpen() &&
+                    <>
+                        <div className="bottom-buttons-group float-end">
+                            <Button variant="outline-secondary"
+                                    className="text-secondary-emphasis me-2"
+                                    disabled={loading}
+                                    onClick={changePullStatus(PullStatus.closed)}>
+                                {loading ?
+                                    <span className="spinner-border spinner-border-sm text-light" role="status"/> :
+                                    <>Close pull request</>
+                                }
+                            </Button>
+                            {!isBranchNotFoundError &&
+                                <Button variant="success"
+                                        disabled={loading || isEmptyDiff}
+                                        onClick={mergePullRequest}>
+                                    {loading ?
+                                        <span className="spinner-border spinner-border-sm text-light" role="status"/> :
+                                        <><GitMergeIcon/> Merge pull request</>
+                                    }
+                                </Button>
                             }
-                        </Button>
-                    }
-                </div>
+                        </div>
+                        {isEmptyDiff &&
+                            <Alert variant="warning" className="mt-4">
+                                Merging is disabled for pull requests without changes.
+                            </Alert>
+                        }
+                    </>
+                }
             </div>
-            <hr className="mt-5 mb-4"/>
-            <CompareBranches
-                repo={repo}
-                reference={{id: pull.destination_branch, type: RefTypeBranch}}
-                compareReference={{id: pull.source_branch, type: RefTypeBranch}}
-            />
+            {isPullOpen() && !isBranchNotFoundError && // in case of branch not found, a different error message is shown
+                <>
+                    <hr className="mt-5 mb-4"/>
+                    <CompareBranches
+                        repo={repo}
+                        reference={{id: pull.destination_branch, type: RefTypeBranch}}
+                        compareReference={{id: pull.source_branch, type: RefTypeBranch}}
+                    />
+                </>
+            }
+            {pull.status === PullStatus.merged && pull.merged_commit_id &&
+                <Link href={{
+                    pathname: '/repositories/:repoId/commits/:commitId',
+                    params: {repoId: repo.id, commitId: pull.merged_commit_id},
+                }}>
+                    Merged commit
+                </Link>
```

Comment:
Can we have something similar to GitHub?
![image](https://github.com/user-attachments/assets/fc167c16-f7d8-4a47-a4d6-ea2c55ed5a09)

Reply:
Yes, though it doesn't have a lot of meaning here.

But it made me notice that header messages were also wrong in case of non-open PRs - 
Fixed these as well.